DXRModules/DeusEx/Classes/DXREventsBase.uc

```diff
@@ -1367,9 +1368,15 @@ function _MarkBingoAsFailed(coerce string eventname)
     } */
 
     data = class'PlayerDataItem'.static.GiveItem(player());
-    if (data.MarkBingoAsFailed(eventname) && class'MenuChoice_ShowBingoUpdates'.static.IsEnabled(dxr.flags)) {
+
+    if (data.MarkBingoAsFailed(eventname)) {
         l(self$"._MarkBingoAsFailed("$eventname$") data: "$data);
-        player().ClientMessage("Failed bingo goal: " $ data.GetBingoDescription(eventname));
+        if (class'MenuChoice_ShowBingoUpdates'.static.MessagesEnabled(dxr.flags) && dxr.localURL != "DX" && dxr.localURL != "DXONLY") {
+            player().ClientMessage("Failed bingo goal: " $ data.GetBingoDescription(eventname));
+        }
+        if (class'MenuChoice_ShowBingoUpdates'.static.SoundsEnabled(dxr.flags) && dxr.localURL != "DX" && dxr.localURL != "DXONLY") {
+            player().PlaySound(Sound'DeusExSounds.Generic.Buzz1', SLOT_None, 3.0);
```

Comment:
```suggestion
        if (last_buzz_time < Level.TimeSeconds-0.1 && class'MenuChoice_ShowBingoUpdates'.static.SoundsEnabled(dxr.flags) && dxr.localURL != "DX" && dxr.localURL != "DXONLY") {
            player().PlaySound(Sound'DeusExSounds.Generic.Buzz1', SLOT_None, 3.0);
            last_buzz_time = Level.TimeSeconds;
```

Reply:
when you define the variable at the top of the file do `var transient float last_buzz_time;`
and then also in `defaultproperties` do `last_buzz_time=-1`

Reply:
next_buzz_time instead of last_buzz_time, that way the extra 0.1 seconds is calculated separately from the if condition

Reply:
I don't see any need to set `nextBuzzTime`'s starting value to -1, unless `Level.TimeSeconds` can ever be negative. If the level time is always nonnegative, then the buzz will always play immediately because it'd check for >= 0, the default value.

Reply:
I actually don't see the need to make `nextBuzzTime` transient. It's preferable that the buzz always plays immediately.

(This is kind of hypothetical though. I can't imagine a scenario where you'd immediately fail a goal by reentering a map. But it's still worth getting right.)

Reply:
> I don't see any need to set `nextBuzzTime`'s starting value to -1, unless `Level.TimeSeconds` can ever be negative. If the level time is always nonnegative, then the buzz will always play immediately because it'd check for >= 0, the default value.

Right you don't need the default anymore due to using nextBuzzTime instead of lastBuzzTime (the offset was the issue before)

Reply:
> I actually don't see the need to make `nextBuzzTime` transient. It's preferable that the buzz always plays immediately.
> 
> (This is kind of hypothetical though. I can't imagine a scenario where you'd immediately fail a goal by reentering a map. But it's still worth getting right.)

transient is what will allow the buzz to play immediately when entering the map, transient means the value is not stored and will be reset to 0.

Reply:
Ah, I had it backward then.